File: src/main/java/org/goafabric/core/data/repository/OrganizationRepository.java
package org.goafabric.core.data.repository;

import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
import io.quarkus.panache.common.Parameters;
import jakarta.enterprise.context.ApplicationScoped;
import org.goafabric.core.data.repository.entity.OrganizationEo;

import java.util.List;

@ApplicationScoped
public class OrganizationRepository implements PanacheRepositoryBase<OrganizationEo, String> {

    public List<OrganizationEo> findByNameStartsWithIgnoreCase(String name) {
        return find("name like concat('%', :name, '%')",
                Parameters.with("name", name)
                        .map()).list();
    }

    public OrganizationEo save(OrganizationEo person) {
        persist(person);
        return person;
    }

}

File: src/main/kotlin/org/goafabric/core/organization/persistence/OrganizationRepository.kt
package org.goafabric.core.organization.persistence

import io.quarkus.hibernate.panache.PanacheRepository
import jakarta.data.repository.Query
import org.goafabric.core.organization.persistence.entity.OrganizationEo

interface OrganizationRepository : PanacheRepository.Managed<OrganizationEo, String> {
    @Query("SELECT o FROM OrganizationEo o WHERE o.name LIKE CONCAT(:name, '%')")
    fun findByNameStartsWith(name: String): List<OrganizationEo>

    fun save(organizationEo: OrganizationEo): OrganizationEo {
        return session.merge(organizationEo)
    }
}
